envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/SanUriMatcherFactory.kt

```diff
@@ -0,0 +1,48 @@
+package pl.allegro.tech.servicemesh.envoycontrol.snapshot.resource.listeners.filters
+
+import io.envoyproxy.envoy.type.matcher.RegexMatcher
+import io.envoyproxy.envoy.type.matcher.StringMatcher
+import pl.allegro.tech.servicemesh.envoycontrol.snapshot.TlsAuthenticationProperties
+import java.lang.IllegalArgumentException
+
+class SanUriMatcherFactory(
+    private val tlsProperties: TlsAuthenticationProperties
+) {
+    private val serviceNameTemplate = "{service-name}"
+    private val sanUriWildcardRegex = escapeRegex(tlsProperties.sanUriFormat)
+
+    private fun escapeRegex(sanUriFormat: String): String {
```

Comment:
`escapeRegex` sounds like some general regex-escaping solution, but this method specifically create `sanUriWildcardRegex `. Maybe name it `creatreSanUriWildcardRegex`?